plugins/plugin-webpack/test/plugin.test.js

```diff
@@ -0,0 +1,31 @@
+const path = require('path');
+const fs = require('fs');
+
+const plugin = require('../plugin');
+
+test('minimal - no options', async () => {
+  const pluginInstance = plugin({
+    buildOptions: {},
+  });
+
+  fs.writeFileSync = jest.fn();
+  fs.writeFile = jest.fn().mockImplementation((...args) => {
+    const callback = args.pop();
+    callback();
+  });
+  console.log = jest.fn();
+
+  await pluginInstance.optimize({
+    buildDirectory: path.resolve(__dirname, 'stubs/minimal/'),
+  });
+
+  expect(fs.writeFileSync.mock.calls.map(toPathAndStringContent)).toMatchSnapshot(
```

Comment:
Nice! I like this approach. The danger with these snapshot tests (sometimes) is that if a file failed to generate, the test would pass because it would only test snapshots for what was generated. But here we’re doing a test specifically to ensure all files are generated before comparing contents, which is perfect.